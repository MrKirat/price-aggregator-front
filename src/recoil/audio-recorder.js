import { atom } from 'recoil';

const isAudioRecorderEnabledState = atom({
    key: 'isAudioRecorderEnabledState',
    default: false
});

export default isAudioRecorderEnabledState;