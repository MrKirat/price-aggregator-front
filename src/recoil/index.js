export { default as isAudioRecorderEnabledState } from './audio-recorder';
export { default as isCameraEnabledState } from './camera';
export { default as searchStringState } from './search';
export { default as productListState } from './products';
export { default as isLoadingState } from './loading';