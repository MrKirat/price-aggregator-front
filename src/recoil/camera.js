import { atom } from 'recoil';

const isCameraEnabledState = atom({
    key: 'isCameraEnabledState',
    default: false
});

export default isCameraEnabledState;