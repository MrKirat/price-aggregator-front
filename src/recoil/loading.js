import { atom } from 'recoil';

const isLoadingState = atom({
    key: 'isLoadingState',
    default: false
});

export default isLoadingState;