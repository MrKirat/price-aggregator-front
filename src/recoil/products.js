import { atom } from 'recoil';

const productListState = atom({
    key: 'productListState',
    default: []
});

export default productListState;