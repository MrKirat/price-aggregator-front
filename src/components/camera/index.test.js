import React from 'react';

import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Camera from './index';
import { RecoilRoot } from 'recoil';

it('renders without crashing', () => {
  const output = mount(<RecoilRoot><Camera /></RecoilRoot>);
  expect(shallowToJson(output)).toMatchSnapshot();
});

it('renders with Camera', () => {
  const output = mount(<RecoilRoot><Camera /></RecoilRoot>);
  expect(output.find('Camera')).toHaveLength(1);
});

it('renders with Webcam', () => {
  const output = mount(<RecoilRoot><Camera /></RecoilRoot>);
  expect(output.find('Webcam')).toHaveLength(1);
});

it('renders with video', () => {
  const output = mount(<RecoilRoot><Camera /></RecoilRoot>);
  expect(output.find('video')).toHaveLength(1);
});

it('renders with icon', () => {
  const output = mount(<RecoilRoot><Camera /></RecoilRoot>);
  expect(output.find('svg')).toHaveLength(1);
});