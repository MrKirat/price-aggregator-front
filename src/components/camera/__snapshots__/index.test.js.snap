// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<RecoilRoot>
  <Batcher
    setNotifyBatcherOfChange={[Function]}
  />
  <Camera>
    <div
      className="makeStyles-container-1"
    >
      <Webcam
        audio={false}
        className="makeStyles-camera-2"
        forceScreenshotSourceSize={false}
        imageSmoothing={true}
        mirrored={false}
        onUserMedia={[Function]}
        onUserMediaError={[Function]}
        screenshotFormat="image/jpeg"
        screenshotQuality={0.92}
      >
        <video
          autoPlay={true}
          className="makeStyles-camera-2"
          muted={false}
          playsInline={true}
          style={Object {}}
        />
      </Webcam>
      <ForwardRef
        className="makeStyles-button-3"
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(SvgIcon))
          className="makeStyles-button-3"
          onClick={[Function]}
        >
          <ForwardRef(SvgIcon)
            className="makeStyles-button-3"
            classes={
              Object {
                "colorAction": "MuiSvgIcon-colorAction",
                "colorDisabled": "MuiSvgIcon-colorDisabled",
                "colorError": "MuiSvgIcon-colorError",
                "colorPrimary": "MuiSvgIcon-colorPrimary",
                "colorSecondary": "MuiSvgIcon-colorSecondary",
                "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                "root": "MuiSvgIcon-root",
              }
            }
            onClick={[Function]}
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-button-3"
              focusable="false"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </ForwardRef(SvgIcon)>
        </WithStyles(ForwardRef(SvgIcon))>
      </ForwardRef>
    </div>
  </Camera>
</RecoilRoot>
`;
